Source/DafnyCore/Triggers/TriggersCollector.cs

```diff
@@ -245,7 +245,7 @@ internal class TriggersCollector {
           expr is SeqSelectExpr ||
           expr is MultiSelectExpr ||
           expr is MemberSelectExpr ||
-          expr is OldExpr ||
+          (expr is OldExpr { Useless: false }) ||
```

Comment:
I don't understand this part.  Why is `old(x)` a trigger candidate when `old` is useful? (Shouldn't just `x` be the trigger?

Reply:
I'm sorry for the piecemeal review.  I need to spend the time to read this carefully (or if you're confident we can just ship it), but I can't do that today :/

Reply:
So I was basically wrong. There was one example of matrices where old(m[i, j]) was not the same trigger as m[i, j] because m was internally using a heap constant. See the error there:

https://github.com/dafny-lang/dafny/actions/runs/3395616639/jobs/5645735703